### PR TITLE
Issue #4557: fixed NPE in NoWhitespaceAfterCheck

### PR DESCRIPTION
**Pull request by sergileon**
Issue [https://github.com/checkstyle/checkstyle/issues/4557](https://github.com/checkstyle/checkstyle/issues/4557)
Fixed NPE and added tests.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java [hunk 1]
```diff
@@ -294,8 +294,15 @@ private static DetailAST getIndexOpPreviousElement(DetailAST ast) {
         else {
             final DetailAST ident = getIdentLastToken(ast);
             if (ident == null) {
+                final DetailAST rparen = ast.findFirstToken(TokenTypes.RPAREN);
+                // construction like new int[]{1}[]
```
Comment (Vladlis) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java]: please use `new int[]{1}[0]` as `new int[]{1}[]` is not compilable

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespaceafter/InputNoWhitespaceAfterArrayDeclarations.java [hunk 2]
```diff
@@ -43,4 +43,10 @@ int get1(int k, int c, int b) [] { //Incorrect
     private long getLongMultArray(int someParam, String value) [][][] { //Incorrect
         return null;
     }
+    int aa = new int[]{1}[0];//Correct
+    int bb = new int[]{1}[ 0];//Incorrect
+    int aaa = new int[][]{{1},{2}}[0][0];//Correct
+    int bbb = new int [][]{{1},{2}}[0][0];//Incorrect
+    int ccc = new int[] []{{1},{2}}[0][0];//Incorrect
+    int ddd = new int[][]{{1},{2}}[0] [0];//Incorrect
```
Comment (Vladlis) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespaceafter/InputNoWhitespaceAfterArrayDeclarations.java]: To be added:
`new int[][]{{1},{2}} [0][0]`
`new int[][]{{1},{2}}[ 0][0]`
`new int[][]{{1},{2}}[0 ][0]`
`new int[][]{{1},{2}}[0][ 0]`
`new int[][]{{1},{2}}[0][0 ]`

**Comment by romani** [comment]
TC and Travis are restarted

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespaceafter/InputNoWhitespaceAfterArrayDeclarations.java (hunk 2 above)
Comment (sergileon) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespaceafter/InputNoWhitespaceAfterArrayDeclarations.java]: I added next tests:
`int in1 = new int[][]{{1},{2}}[ 0][0];//Correct`
`int in2 = new int[][]{{1},{2}}[0 ][0];//Correct`
`int in3 = new int[][]{{1},{2}}[0][ 0];//Correct`
`int in4 = new int[][]{{1},{2}}[0][0 ];//Correct`

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespaceafter/InputNoWhitespaceAfterArrayDeclarations.java [hunk 3]
```diff
@@ -43,4 +43,6 @@ int get1(int k, int c, int b) [] { //Incorrect
     private long getLongMultArray(int someParam, String value) [][][] { //Incorrect
         return null;
     }
+    int aa = new int[]{1}[0];//Correct
+    int bb = new int[]{1}[ 0];//Incorrect
```
[the remark this answers is not in the document]
Comment (sergileon) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespaceafter/InputNoWhitespaceAfterArrayDeclarations.java]: I added next tests:
`int aaa = new int[][]{{1},{2}}[0][0];//Correct`
`int bbb = new int [][]{{1},{2}}[0][0];//Incorrect`
`int ccc = new int[] []{{1},{2}}[0][0];//Incorrect`
`int ddd = new int[][]{{1},{2}} [0][0];//Incorrect`
`int eee = new int[][]{{1},{2}}[0] [0];//Incorrect`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java [hunk 4]
```diff
@@ -162,7 +162,13 @@ private static DetailAST getWhitespaceFollowedNode(DetailAST ast) {
                 whitespaceFollowedAst = getArrayDeclaratorPreviousElement(ast);
                 break;
             case TokenTypes.INDEX_OP:
-                whitespaceFollowedAst = getIndexOpPreviousElement(ast);
+                final DetailAST resultAst = getIndexOpPreviousElement(ast);
+                if (resultAst == null) {
```
[the remark this answers is not in the document]
Comment (sergileon) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java]: Moved to `getIndexOpPreviousElement`

**Comment by romani** [comment]
we released 8.2 version, please rebase all your PRs to our latest master to avoid CI problems

**Comment by sergileon** [comment]
report on the diff [https://sergileon.github.io/](https://sergileon.github.io/)

**Comment by romani** [comment]
>lucene-solr (3)

Just another example that our regression config is incomplete and miss a lot of cases that are possible to catch without waiting for issue to be opened.

**Comment by Vladlis** [comment]
@romani 
take a look on the issue description - bug is found by CI

**Pull request merged**